grove/plugin/jwt_auth.go

```diff
@@ -0,0 +1,162 @@
+package plugin
+
+/*
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/apache/trafficcontrol/lib/go-log"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/dgrijalva/jwt-go/request"
+	"github.com/iancoleman/strcase"
+)
+
+type JwtSecret struct {
+	Issuer   string `json:"iss"`
+	Audience string `json:"aud"`
+	Algoritm string `json:"alg"`
+	Kid      string `json:"kid"`
+	Value    string `json:"value"`
+}
```

Comment:
Consider using a JWK for the Algorithm, Kid, and Value. That will give you considerably more flexibility in the future as additional algorithms become desirable. There's a good JWK library I've used for this purpose here: https://godoc.org/github.com/lestrrat/go-jwx/jwk . (It's got a good JWT library next to it that does some of the heavy lifting if you wanted to do some heavier refactoring; though I don't think that's necessarily important for this.)

Using the standard format for JWKs will work nicely with the JSON-based config, too, since it's a standard way of formatting the keys that users of JWTs will expect and understand.